doc/design/custom_xcode_schemes.md

```diff
@@ -0,0 +1,501 @@
+    schemes = _SCHEMES,
+    targets = [
+        "//Sources/Foo",
+    ],
+)
+```
+
+#### Collecting Targets from Schemes
+
+Let's look at an example that illustrates how targets are collected. The following is the example
+from earlier with two schemes and some overlapping targets.
+
+```python
+# Assumptions
+#   //Sources/Foo:Foo - swift_library
+#   //Sources/FooTests:FooTests = ios_unit_test
+#   //Sources/App = ios_application
+#   //Sources/AppUITests = ios_ui_test
+
+load(
+    "@com_github_buildbuddy_io_rules_xcodeproj//xcodeproj:xcodeproj.bzl",
+    "xcode_schemes",
+    "xcodeproj",
+)
+
+_SCHEMES = [
+    xcode_schemes.scheme(
+        name = "Foo Module",
+        build_action = xcode_schemes.build_action(["//Sources/Foo"]),
+        test_action = xcode_schemes.test_action(["//Tests/FooTests"]),
+    ),
+    xcode_schemes.scheme(
+        name = "My Application",
+        test_action = xcode_schemes.test_action([
+            "//Tests/AppUITests",
+            "//Tests/FooTests",
+        ]),
+        launch_action = xcode_schemes.launch_action("//Sources/App"),
+    ),
+]
+
+xcodeproj(
+    name = "generate_xcodeproj",
+    project_name = "Command Line",
+    tags = ["manual"],
+    schemes = _SCHEMES,
+)
+```
+
+After evaluation of the macro, the declaration for `_xcodeproj` will look like the following:
+
+```python
+_xcodeproj(
+    name = "generate_xcodeproj",
+    project_name = "Command Line",
+    tags = ["manual"],
+    schemes = [
+        "{...}", # JSON representation of Foo Module scheme
+        "{...}", # JSON representation of My Application scheme
+    ],
+    # Only the top-level or leaf node targets were added to the overall targets 
+    # list.
+    targets = [
+        "//Sources/App",
+        "//Tests/FooTests",
+        "//Tests/AppUITests",
+    ],
+)
+```
+
+Note that the `targets` list is a deduplicated, sorted list of the targets that were provided from
+the schemes and directly in the `targets` parameter.
+
+The JSON representation of `Foo Module` will look like the following:
+
+```json
+{
+  "name": "Foo Module",
+  "build_action": {
+    "targets": [
+      "//Sources/Foo",
+      "//Tests/FooTests"
+    ],
+  },
+  "test_action": {
+    "targets": [
+      "//Tests/FooTests"
+    ]
+  }
+}
+```
+
+Note that the build action targets list contains the targets from the build action declaration and
+the test action declaration.
+
+The JSON representation of `My Application` will look like the following:
+
+```json
+{
+  "name": "My Application",
+  "build_action": {
+    "targets": [
+      "//Sources/App",
+      "//Tests/FooTests"
+      "//Tests/AppUITests"
+    ],
+  },
+  "test_action": {
+    "targets": [
+      "//Tests/FooTests",
+      "//Tests/AppUITests"
+    ]
+  },
+  "launch_action": {
+    "target": "//Sources/App"
+  }
+}
+```
+
+> TODO: Is selecting targets from the test actions and launch actions sufficient for only incuding
+> top-level targets in the overall targets list?
```

Comment:
What do you mean by this?

Reply:
In short, are you OK with the technique of populating the the `targets` list for `_xcodeproj` rule using the targets listed under `test_action` and `launch_action`?

Reply:
Yes.